lib/location/decorate.mjs

```diff
@@ -42,18 +42,18 @@ async function update() {
 
   const newValues = {}
 
-  this.infoj
-    .filter(entry => typeof entry.newValue !== 'undefined')
-    .forEach(entry => {
+  for (const entry of this.infoj.filter(entry => typeof entry.newValue !== 'undefined')) {
```

Comment:
Recommend separating filtering array to a descriptive variable prior to the for loop for better readability